### PR TITLE
[IVY] "local" compiler

### PR DESCRIPTION
**Pull request by vicb**

### CODE REVIEW

File: packages/compiler/src/render3/r3_view_compiler_local.ts [hunk 1]
```diff
@@ -649,194 +622,173 @@ class TemplateDefinitionBuilder implements TemplateAstVisitor, LocalResolver {
...
+    const attributeMap: {[name: string]: string} = {};
+
+    template.attributes.forEach(a => {
+      attributeNames.push(asLiteral(a.name), asLiteral(''));
+      attributeMap[a.name] = a.value;
     });
+
+    // Match directives on template attributes
+    if (this.directiveMatcher) {
+      const selector = createCssSelector('ng-template', attributeMap);
```
Comment (alxhub) [inline packages/compiler/src/render3/r3_view_compiler_local.ts]: We should cache these so when parsing different templates we don't re-parse the same selectors.

File: packages/compiler/test/render3/r3_compiler_compliance_spec.ts [hunk 2]
```diff
@@ -1156,14 +1156,14 @@ describe('compiler compliance', () => {
             template: function MyComponent_Template(rf: IDENT, ctx: IDENT) {
               if (rf & 1) {
                 $r3$.ɵE(0, 'ul');
-                $r3$.ɵC(1, MyComponent_ForOfDirective_Template_1, null, $_c0$);
+                $r3$.ɵC(1, MyComponent_li_Template_1, null, $_c0$);
```
Comment (alxhub) [inline packages/compiler/test/render3/r3_compiler_compliance_spec.ts]: Could we know that the template came from a `*ngFor` binding and include that information in the name?

**Review by alxhub — APPROVED** [review]

**Review by alxhub — APPROVED** [review]

**Comment by vicb** [comment]
http://test/OCL:193572350:BASE:193587591:1524178082713:95eb9775

**Comment by vicb** [comment]
submitting:
- lint is wrong,
- pull approve is weird.

**Pull request closed without merging**